### ISSUE TITLE
database error 

### ISSUE THREAD

**Issue by h09s** [UnamSanctam/UnamWebPanel#338]
so its me again iam getting this error now 
![image](https://github.com/UnamSanctam/UnamWebPanel/assets/72148172/3a9f4450-4be0-4234-80f2-8c1a487fad94)
it has the write perms and everything

**Comment by UnamSanctam** [comment]
You would need to check the error log for that, you can also try the UnamWebPanel 1.8.0 BETA: [UnamWebPanel 1.8.0 BETA2.zip](https://github.com/UnamSanctam/UnamWebPanel/files/14184110/UnamWebPanel.1.8.0.BETA2.zip) which is technically finished, I'm just waiting to see if some testers find any bugs.

**Comment by UnamSanctam** [comment]
Those might be old errors, but if they are new ones then the database isn't actually writable at the moment, recheck the permissions again.

**Comment by UnamSanctam** [comment]
And what about the `db` folder and `unamwebpanel.db` file?

**Comment by UnamSanctam** [comment]
And if you remove the error log then the same error starts appearing?

**Comment by Demon4u** [comment]
error 500. I don't like the web.config code
```
<httpProtocol>
      <customHeaders>
        <add name="X-Robots-Tag" value="noindex, nofollow" />
      </customHeaders>
    </httpProtocol>
```
if you delete this piece, the panel starts.
and configuration changes are not saved.

**Comment by UnamSanctam** [comment]
Aha, you're using IIS, haven't tested it yet. That part is to tell search engines not to crawl and index the pages.

**Comment by Demon4u** [comment]
In IIS, the error sounds like: the "httpprotocol" configuration section cannot be read due to a missing section declaration

**Comment by UnamSanctam** [comment]
I think it should be in the system.webServer and not the system.web section, can you try with this?

```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <system.web>
    <globalization requestEncoding="utf-8" responseEncoding="utf-8"/>
  </system.web>
  <system.webServer>
    <httpProtocol>
      <customHeaders>
        <add name="X-Robots-Tag" value="noindex, nofollow"/>
      </customHeaders>
    </httpProtocol>
  </system.webServer>
  <location path="db">
    <system.web>
      <authorization>
        <deny users="*"/>
      </authorization>
    </system.web>
  </location>
</configuration>
```

**Comment by Demon4u** [comment]
> I think it should be in the system.webServer and not the system.web section, can you try with this?
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <configuration>
>   <system.web>
>     <globalization requestEncoding="utf-8" responseEncoding="utf-8"/>
>   </system.web>
>   <system.webServer>
>     <httpProtocol>
>       <customHeaders>
>         <add name="X-Robots-Tag" value="noindex, nofollow"/>
>       </customHeaders>
>     </httpProtocol>
>   </system.webServer>
>   <location path="db">
>     <system.web>
>       <authorization>
>         <deny users="*"/>
>       </authorization>
>     </system.web>
>   </location>
> </configuration>
> ```

Yes, it worked.

**Comment by UnamSanctam** [comment]
>Yes, it worked.

Great, thanks for reporting the problem.

**Comment by Demon4u** [comment]
![Screenshot_1](https://github.com/UnamSanctam/UnamWebPanel/assets/85320553/9d472423-fdc0-430c-b7d9-d36e0c101b63)
when changing the configuration

**Comment by UnamSanctam** [comment]
>when changing the configuration

Because that's incorrect JSON. Something like `{ "test": true }` should work.

**Comment by Demon4u** [comment]
I wrote it wrong.
she writes that it is saved, but when the page is refreshed, it is empty again

**Comment by UnamSanctam** [comment]
>I wrote it wrong.
she writes that it is saved, but when the page is refreshed, it is empty again

Hmm it's working for me, do you get any errors in your console? Could be some difference due to IIS.

**Comment by Demon4u** [comment]
![Screenshot_1](https://github.com/UnamSanctam/UnamWebPanel/assets/85320553/db875ee1-43f4-41e1-b9d7-b016c2da8f69)
Moreover, when adding a new configuration, it appears, but also without parameters

There are no errors in the console

**Comment by UnamSanctam** [comment]
Are there any error logs anywhere? Like inside UNAM_LIB\Logs? I will have to set up an IIS server later and test I guess.

**Comment by Demon4u** [comment]
> Are there any error logs anywhere? Like inside UNAM_LIB\Logs? I will have to set up an IIS server later and test I guess.

No, there is no error in any log

**Comment by UnamSanctam** [comment]
>No, there is no error in any log

Alright, then I guess I'll have to set up an IIS server later to test when I have the time.

**Comment by Demon4u** [comment]
> > No, there is no error in any log
> 
> Alright, then I guess I'll have to set up an IIS server later to test when I have the time.

I opened the database file and saw that all the configurations are there, and if the worker chooses a new configuration, then it works. Although the configuration options panel itself is empty

**Comment by UnamSanctam** [comment]
>I opened the database file and saw that all the configurations are there, and if the worker chooses a new configuration, then it works. Although the configuration options panel itself is empty

Hmm, I just installed IIS and set up the web panel, the configuration page is working fine for me. Are you sure that you're using the UnamWebPanel 1.8.0 BETA2 that I sent previously in this issue?

**Comment by Demon4u** [comment]
> Hmm, I just installed IIS and set up the web panel, the configuration page is working fine for me. Are you sure that you're using the UnamWebPanel 1.8.0 BETA2 that I sent previously in this issue?

Yes, that's what I installed

**Comment by UnamSanctam** [comment]
>Yes, that's what I installed

Then that's strange, does the same happen with this one: [UnamWebPanel 1.8.0 BETA3.zip](https://github.com/UnamSanctam/UnamWebPanel/files/14186416/UnamWebPanel.1.8.0.BETA3.zip)?

>I use the one in release page

Yes that's fine @Demon4u is testing the Beta of the next version.

**Comment by Demon4u** [comment]
> > Yes, that's what I installed
> 
> Then that's strange, does the same happen with this one: [UnamWebPanel 1.8.0 BETA3.zip](https://github.com/UnamSanctam/UnamWebPanel/files/14186416/UnamWebPanel.1.8.0.BETA3.zip)?
> 
> > I use the one in release page
> 
> Yes that's fine @Demon4u is testing the Beta of the next version.

Beta3 - everything is fine!

**Comment by UnamSanctam** [comment]
>Beta3 - everything is fine!

Alright great! Thank you for the tests and reports! I'm probably planning on releasing it today.

**Comment by Brovkines** [comment]
> > Beta3 - everything is fine!
> 
> Alright great! Thank you for the tests and reports! I'm probably planning on releasing it today.

Hello. When will BETA3 be released and available?

**Comment by alcinzal** [comment]
> Hello. When will BETA3 be released and available?

v1.8.0 is already released. Check it out: https://github.com/UnamSanctam/UnamWebPanel/releases

**Comment by Brovkines** [comment]
> > Hello. When will BETA3 be released and available?
> 
> v1.8.0 is already released. Check it out: https://github.com/UnamSanctam/UnamWebPanel/releases

Is this the latest and most "secure" version?

**Comment by alcinzal** [comment]
> Is this the latest and most "secure" version?

Yes, v1.8.0 is the latest and most secure version. If you read through the changelog you can see some points about improving the security of the panel, like:
* Added constant verification of the hashed password during login sessions, ensuring that any change to the password results in all users being logged out
* Added further XSS mitigation to both the endpoint, the pages, the APIs and the datatable server side output
* Added many strict headers that improve browser security

**Comment by VK-VZ** [comment]
Hello.  How to transfer workers from the old panel?  Just move the db folder?  And if so, then in the new version you did not modify the .db file.  Is it correct to replace the new .db with the old one?

**Comment by alcinzal** [comment]
> Hello. How to transfer workers from the old panel? Just move the db folder? And if so, then in the new version you did not modify the .db file. Is it correct to replace the new .db with the old one?

Unam did change the database file in the new update. He added the tables ipblocking and loginblock, maybe some more stuff aswell. Although you might not keep your statistics, the workers will automatically be added to the new database when they connect. However before you do anything, we should probably wait for an answer from Unam just to be safe.

**Comment by VK-VZ** [comment]
> > Beta3 - everything is fine!
> 
> Alright great! Thank you for the tests and reports! I'm probably planning on releasing it today.

unam please answer the question at the end. About transferring from the old panel

**Comment by Demon4u** [comment]
> > > Beta3 - everything is fine!
> > 
> > 
> > Alright great! Thank you for the tests and reports! I'm probably planning on releasing it today.
> 
> unam please answer the question at the end. About transferring from the old panel

They will connect to the new panel themselves. You don't need to transfer anything

**Comment by UnamSanctam** [comment]
>unam please answer the question at the end. About transferring from the old panel

Either replace the old db with the new db file, the miner will automatically get added the next time they connect. But if you need to keep your old configurations and statistics then it's possible for you to open the old db in any SQLite Database Browser and export all the data (only the data, not the structures), and then open the new db and import the data.
